arch/riscv/src/lib.rs

```diff
@@ -45,105 +45,102 @@ extern "C" {
     static __global_pointer: usize;
 }
 
+/// Entry point of all programs (`_start`).
+///
+/// This assembly does three functions:
+///
+/// 1. It initializes the stack pointer, the frame pointer (needed for closures
+///    to work in start_rust) and the global pointer.
+/// 2. It initializes the .bss and .data RAM segments. This must be done before
+///    any Rust code runs. See https://github.com/tock/tock/issues/2222 for more
+///    information.
+/// 3. Finally it calls `main()`, the main entry point for Tock boards.
 #[cfg(any(doc, all(target_arch = "riscv32", target_os = "none")))]
-extern "C" {
-    // Entry point of all programs (`_start`).
-    ///
-    /// This assembly does three functions:
-    ///
-    /// 1. It initializes the stack pointer, the frame pointer (needed for closures
-    ///    to work in start_rust) and the global pointer.
-    /// 2. It initializes the .bss and .data RAM segments. This must be done before
-    ///    any Rust code runs. See <https://github.com/tock/tock/issues/2222> for more
-    ///    information.
-    /// 3. Finally it calls `main()`, the main entry point for Tock boards.
-    pub fn _start();
+#[link_section = ".riscv.start"]
+#[export_name = "_start"]
+#[unsafe(naked)]
+pub extern "C" fn _start() {
+    use core::arch::naked_asm;
+    naked_asm! ("
+        // Set the global pointer register using the variable defined in the
+        // linker script. This register is only set once. The global pointer
+        // is a method for sharing state between the linker and the CPU so
+        // that the linker can emit code with offsets that are relative to
+        // the gp register, and the CPU can successfully execute them.
```

Comment:
Did rustfmt change this indentation?

Reply:
I think I changed it manually because one level in from the naked_asm! indentation seemed right